lib/publiccloud/provider.pm

```diff
@@ -451,8 +451,13 @@ sub terraform_apply {
     my $resource_id;
     if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
         foreach my $vm_type ('cluster_nodes', 'drbd', 'netweaver') {
-            push @{$vms}, @{$output->{$vm_type . '_name'}->{value}};
-            push @{$ips}, @{$output->{$vm_type . '_public_ip'}->{value}};
+            if (is_azure && $vm_type eq 'cluster_nodes') {
+                push @{$vms}, @{$output->{$vm_type . '_name'}->{value}[0]};
+                push @{$ips}, @{$output->{$vm_type . '_public_ip'}->{value}[0]};
```

Comment:
Did you confirm this works? Somehow  I think it should be `->[0]`

Reply:
> Did you confirm this works? Somehow I think it should be `->[0]`

yes

Reply:
I don't see it in the VR, and a quick test makes me think it will not work:
```
my @arr = qw(hola mundo);
my $r = \@arr;
print "Sin flecha: $r[0]\n";
print "Con flecha: $r->[0]\n";
Sin flecha: 
Con flecha: hola
```

Reply:
Tested with the JSON output and `decode_json` and confirmed it works without the `->[0]`.